### DISCUSSION TITLE
x-mask:dynamic="$money($input)" cannot auto re-format digit group character separator

### DISCUSSION THREAD

**Discussion by bayukartiko** [alpinejs/alpine#2979] [7. Bugs (archived)]
Hi, I discovered a bug or problem that when I input in the 'cost' field with `x-mask:dynamic='$money($input)'` that works but not what I expected. 
when I type 'for example `USD 10,000`' that works fine BUT when I delete `0` on the last character, I want `x-mask` to become `USD 1,000` and the problem is ` x-mask` only deletes that and will become `USD 10,00` that result is not what I expected. (**Note**: I did not set the decimal character separator).

I think you should see [`autoNumeric`](https://github.com/autoNumeric/autoNumeric) library for reference to my question.

**Comment by Nielssie** [comment]
Noticed the same problem. When you 'leave' the input field the reformatting is correct. 
Do you already have a solution or work around?
https://alpinejs.dev/plugins/mask

**Comment by bayukartiko** [comment]
@Nielssie I know it, but what I want is reformating when the user is `still typing` in that input field, and **NOT** `after leaving` the input field.

that's why in my question I inserted a [`autoNumeric`](https://github.com/autoNumeric/autoNumeric) library for reference.
